src/main/java/io/github/connorhartley/guardian/internal/detections/FlyDetection.java

```diff
@@ -161,139 +168,229 @@ public Guardian getPlugin() {
-                    punishmentLevels, new TypeToken<Map<String, Double>>() {
-            });
-
-            HashMap<String, String> punishmentProperties = new HashMap<>();
-            punishmentProperties.put("channel", "admin");
-            punishmentProperties.put("releasetime", "12096000");
-
-            this.configPunishmentProperties = new StorageValue<>(new StorageKey<>("punishment-properties"),
-                    "Properties that define certain properties for all the punishments in this detection.",
-                    punishmentProperties, new TypeToken<Map<String, String>>() {
-            });
-
-            HashMap<String, List<String>> customPunishments = new HashMap<>();
-            customPunishments.put("example", Collections.singletonList("msg %player% You have been prosecuted for illegal action!"));
-
-            this.configCustomPunishments = new StorageValue<>(new StorageKey<>("custom-punishments"),
-                    "Custom punishments that can execute custom commands.",
-                    customPunishments, new TypeToken<Map<String, List<String>>>() {
-            });
-
-            HashMap<String, Double> severityDistribution = new HashMap<>();
-            severityDistribution.put("lower", 0d);
-            severityDistribution.put("mean", 10d);
-            severityDistribution.put("standard-deviation", 5d);
-
-            this.configSeverityDistribution = new StorageValue<>(new StorageKey<>("severity-distribution"),
-                    "Normal distribution properties for calculating the over-shot value from the mean.",
-                    severityDistribution, new TypeToken<Map<String, Double>>() {
-            });
+        @Override
+        public void load() {
+            try {
+                if (this.exists()) {
+                    this.configurationNode = this.configManager.load(this.flyDetection.getPlugin().getConfigurationOptions());
+
+                    this.configAnalysisTime.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configAltitudeMaximum.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configTickBounds.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configPunishmentLevels.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configPunishmentProperties.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configCustomPunishments.<ConfigurationNode>loadStorage(this.configurationNode);
+                    this.configSeverityDistribution.<ConfigurationNode>loadStorage(this.configurationNode);
+
+                    this.configManager.save(this.configurationNode);
+                }
+            } catch (IOException e) {
+                this.flyDetection.getPlugin().getLogger().error("A problem occurred attempting to load Guardians global configuration!", e);
```

Comment:
Also the fact that it is talking about the "global configuration" when it is only working with the detection's configuration.